zookeeper-api/src/main/java/org/apache/helix/zookeeper/zkclient/ZkClient.java

```diff
@@ -1858,24 +1923,47 @@ public Object call() throws Exception {
     });
   }
 
+  private boolean watchForData(final String path, boolean skipWatchingNodeNotExist) {
+    try {
+      if (skipWatchingNodeNotExist) {
+        Stat stat = new Stat();
```

Comment:
getData will return stat too. 
`retryUntilConnected(() -> (((ZkConnection) getConnection()).getZookeeper().getData(path, true, stat)));`

Here, we can't pass null to getData(). It would throw exception when getData try to fill data into stat.

Reply:
@kaisun2000 `null` is acceptable for the param `stat`. It would not throw exception. See [here](https://github.com/apache/zookeeper/blob/b0aaa7de5b28375cd34ead8c8fbf14e3ebddd30f/src/java/test/org/apache/zookeeper/test/AuthTest.java#L70)
```
    public void testBadAuthThenSendOtherCommands() throws Exception {
        ZooKeeper zk = createClient();     
        try {        
            zk.addAuthInfo("INVALID", "BAR".getBytes());
            zk.exists("/foobar", false);             
            zk.getData("/path1", false, null);
            Assert.fail("Should get auth state error");
```